src/roomba_robot/botgal_and_lib/main.c

```diff
@@ -0,0 +1,196 @@
+    printf("%d,%d,%s", get_create_rflightbump(), 0, "right light front bumper \n");
+    printf("%d,%d,%s", get_create_rfcliff(), 0, "right cliff \n");
+    printf("%d,%d,%s", get_create_rclightbump(), 0, "right cliff light bumper \n");
+    printf("%d,%d,%s", get_create_cwdrop(), 0, "front drop wheel \n");
+    printf("%d,%d,%s", get_create_lclightbump(), 0, "left cliff light bumper \n");
+    printf("%d,%d,%s", get_create_lfcliff(), 0, "left front cliff \n");
+    printf("%d,%d,%s", get_create_lflightbump(), 0, "left light bumper \n");
+    printf("%d,%d,%s", get_create_lbump(), 0, "left bumper \n");
+    printf("%d,%d,%s", get_create_llightbump(), 0, "left light bumper \n");
+    printf("%d,%d,%s", get_create_lcliff(), 0, "left cliff \n");
+    printf("%d,%d,%s", get_create_lwdrop(), 0, "left wheel drop \n");
+    printf("%d,%d,%s", get_create_distance(), 0, "total distance\n");
+    printf("%d,%d,%s", get_create_total_angle(), 0, "total angle \n");
+}
+
+bool setupcamera() {
+    if (camera_open() == 0 || camera_update() == 0) return false;
+    return true;
+}
+
+void setuprobot(bool debug, bool camera) {
+    create_connect();
+    setupsensordata();
+    if (debug == true) {
+        printsensordata();
+    }
+    if (camera) {
+        if (setupcamera()) printf("%s", "Camera initialized");
+        else printf("%s", "Camera Error");
+    }
+    enable_servos();
+}
+
+void shutdownrobot(bool camera) {
+    if (camera) {
+        camera_close();
+    }
+    disable_servos();
+    ao();
+    create_stop();
+    create_disconnect();
+}
+
+void drivestraight(int speed, bool direction, int distance,
+                   bool watch_for_touch) { //false backwards true forwards distance in mm
+    if (direction == true) {
+        set_create_distance(0);
+        while (get_create_distance() <= (distance)) {
+            create_drive_direct(speed, speed);
+            if (watch_for_touch) {
+                if (get_create_lbump() || get_create_rbump() == true) {
+                    set_create_distance(distance);
+                }
+            }
+        }
+    } else {
+        set_create_distance(0);
+        while ((-1 * get_create_distance()) <= (distance)) {
+            create_drive_direct(speed * -1, speed * -1);
+        }
+    }
+}
+
+void turn(int speed, int deg) {
+    // deg - nach rechts + nach links
+    set_create_distance(0);
+    set_create_total_angle(0);
+    if (deg > 0) {
+        while (get_create_total_angle() <= deg) {
+            create_drive_direct(speed * -1, speed);
+        }
+    } else {
+        while (get_create_total_angle() >= deg) {
+            create_drive_direct(speed, speed * -1);
+        }
+    }
+}
+
+void wasd() { // function to control robot via webinterface
+    bool exit;
+    exit = true;
+    int speed = 100;
+    while (exit) {
+        char input;
+        printf("%s", "Please provide input:\n");
+        scanf("%c", &input);
+        if (input == 'w') {
```

Comment:
Switch-case instead of if-statements!

```c
switch (expression)
​{
    case constant1:
      // statements
      break;

    case constant2:
      // statements
      break;
    .
    .
    .
    default:
      // default statements
}
```